lib/expression/callbacks/standard.ex

```diff
@@ -145,6 +145,28 @@ defmodule Expression.Callbacks.Standard do
     }
   end
 
+  @doc """
+  Parse random dates and times with `strftime` patterns and return a DateTime value
+  when it matches.
+  """
+  @expression_doc doc: "Parse a date value using strftime formatting and return a DateTime",
+                  expression: "parse_datevalue(\"2016-02-29T22:25:00-00:00\", \"%FT%T%:z\")",
+                  result: DateTime.new!(~D[2016-02-29], ~T[22:25:00])
+  @expression_doc doc: "Attempt to parse a date value and return nil when failing",
```

Comment:
are these doctested? If not could we add tests?

Reply:
these @expression_docs are compiled by the macro to become doctests and documentation at the same time

Reply:
really impressive, so cool!